### DISCUSSION TITLE
Ambiant dialogue condition that indicates a back attack battle

### DISCUSSION THREAD

**Discussion by Algarad** [Albeoris/Memoria#807] [Q&A] [answered]
Hi all,

so here we are with my monthly requests for help for ambiant dialogues.

My characters plays audio at the start of the battle taunting the enemies or whatever, but they also do it when the battle start with a back attack and of course they sound stupid when it happens!
In the BattleCalculator.cs, under "public static class BattleState" there a set of conditions that indicates friendly battles, ragtime battles and whatsoever, but there is no _isBackAttackBattle_ condition.

I noticed that there is "IsCasterVisuallySameDirectionTarget()", but I do not understand how it should be used and I guess it is not useful for what I want to achieve anyway, since it talks about a "caster" and no one is casting anything.

Any idea?

Thank you for your support!

**Comment by Tirlititi** [comment, marked as answer]
Hello.

Don't you use [``BattleVoiceEffects.txt``](https://github.com/Albeoris/Memoria/wiki/Voice-acting-system#user-content-Ambiant_dialogs) for that kind of things? It's meant to handle all these ambiant dialogs and it uses NCalc formulas instead of C# code.
With NCalc formulas, you can use ``IsBattlePreemptive`` and ``IsBattleBackAttack``.

In terms of C#, you would use:
```csharp
if (FF9StateSystem.Battle?.FF9Battle?.btl_scene?.Info != null && FF9StateSystem.Battle.FF9Battle.btl_scene.Info.StartType == battle_start_type_tags.BTL_START_FIRST_ATTACK)
    // Current battle started with a Preemptive strike
if (FF9StateSystem.Battle?.FF9Battle?.btl_scene?.Info != null && FF9StateSystem.Battle.FF9Battle.btl_scene.Info.StartType == battle_start_type_tags.BTL_START_BACK_ATTACK)
    // Current battle started with a Back attack
```

**Comment by Algarad** [comment]
Yes... Okay, I'm definitely blind. I was looking in the wrong place.
Sorry if I abused your patience with a stupid question, I should have checked the wiki better.

Thank you!

**Reply by Tirlititi** [reply]
No problem; issues appear to be "stupid" only after you overcome them ^^
